lib/rspec/rails/matchers/have_enqueued_mail.rb

```diff
@@ -88,6 +88,18 @@ def arguments_match?(job)
           super(job)
         end
 
+        def process_arguments(job, given_mail_args)
+          if job[:job] == ActionMailer::MailDeliveryJob
```

Comment:
What if it's a custom delivery job set with [`config.action_mailer.delivery_job`](https://guides.rubyonrails.org/configuring.html)?

Reply:
If job is a subclass of `ActionMailer::MailDeliveryJob` the code should work, maybe I can replace with `job[:job] <= ActionMailer::MailDeliveryJob` to also catch subclasses.

Also I'll add a `defined?(ActionMailer::MailDeliveryJob)` to avoid to break when running in rails < 6.0.